test/util/fake-timers-test.js

```diff
@@ -102,72 +102,84 @@ describe("fakeTimers.clock", function() {
             this.clock = fakeTimers.clock.create();
         });
 
-        it("returns numeric id or object with numeric id", function() {
-            var result = this.clock.setImmediate(function() {
-                return;
-            });
+        if (typeof setImmediate === "function") {
```

Comment:
I think we elsewhere in the codebase agreed to use the more Mocka-ish way of using `this.skip()` (`grep -C5 this.skip test/*.js` for hits). Could be used in both `before*()` and in `it()`. A la

```
252-        before(function() {
253-            if (!global.Promise) {
254:                this.skip();
255-            }
256-        });
```

Reply:
Yes, I think we agree on that. We should probably refactor the file to use skip by grouping the corresponding tests in contexts.

However, the rest of this specific file is using if-statements and I aligned with that.